pkg/resources/monitoring/cortex/helpers.go

```diff
@@ -547,14 +547,24 @@ func AddServiceMonitor() CortexServiceOption {
 	}
 }
 
+func serviceOverride(componentOverride string, fallback string) string {
+	if componentOverride != "" {
+		return componentOverride
+	}
+	return fallback
+}
+
 func (r *Reconciler) buildCortexWorkloadServices(
+	componentOverride,
```

Comment:
It might be better if this was an option, like this:

```go
func (r *Reconciler) buildCortexWorkloadServices(
	target string,
	opts ...CortexServiceOption,
) []resources.Resource {
	options := CortexServiceOptions{
    targetLabels:  cortexWorkloadLabels(target)
  }
	options.apply(opts...)
}
```
```go
func (r *Reconciler) services() []resources.Resource {
	resources := []resources.Resource{
		r.memberlistService(),
	}
  svcOpts := []CortexServiceOptions{}
  if targets := r.mc.Spec.Cortex.CortexWorkloads.GetTargets(); len(targets) == 1 && targets[0] == "all" {
    svcOpts = append(svcOpts, WithTargetLabels(cortexWorkloadLabels("all")))
  }
	resources = append(resources, r.buildCortexWorkloadServices(alertmanager,
		append(svcOpts, AddServiceMonitor())...,
	)...)
	resources = append(resources, r.buildCortexWorkloadServices(compactor,
		append(svcOpts, AddServiceMonitor())...,
	)...)
```